thorlcr/graph/thgraphslave.cpp

```diff
@@ -949,8 +949,18 @@ bool CSlaveGraph::recvActivityInitData(size32_t parentExtractSz, const byte *par
 
     if (syncInitData())
     {
-        if (!graphCancelHandler.recv(queryJobChannel().queryJobComm(), msg, 0, mpTag, NULL, LONGTIMEOUT))
-            throw MakeStringException(0, "Error receiving actinit data for graph: %" GIDPF "d", graphId);
+        CTimeMon timer;
+        while (!graphCancelHandler.recv(queryJobChannel().queryJobComm(), msg, 0, mpTag, NULL, MEDIUMTIMEOUT))
```

Comment:
If checking for isCancelled in loop, why do it at MEDIUMTIMEOUT (30000) interval ?  Why not at 1000 or so to end without delay ?

Reply:
The loop is here mainly to issue a line of tracing every interval, and we wouldn't want that every 1 second. If there's an abort condition it won't wait that long, the recv will be cancelled.

Reply:
ok.  I like small granularity for knowing it was cancelled, and then tracing only every 30 iterations, etc.  But no matter, just my approach.